tests/js/searchtools.js

```diff
@@ -55,18 +44,11 @@ describe('Basic html theme search', function() {
     });
 
     it('should partially-match "sphinx" when in title index', function() {
-      index = {
-        docnames:["index"],
-        filenames:["index.rst"],
-        terms:{'useful': 0, 'utilities': 0},
-        titles:["sphinx_utils module"],
-        titleterms:{'sphinx_utils': 0}
-      }
-      Search.setIndex(index);
-      searchterms = ['sphinx'];
-      excluded = [];
-      terms = index.terms;
-      titleterms = index.titleterms;
+      eval(loadFixture("partial/searchindex.js"));
+
+      [/* first-ignored */, searchterms, excluded, /*rest-ignored*/] = Search._parseQuery('sphinx');
```

Comment:
I would personally use underscores in these cases but it's not that big a deal.

```suggestion
      [_, searchterms, excluded, _] = Search._parseQuery('sphinx');
```

Reply:
JavaScript's unpacking behaviour here seems somewhat inconsistent to me; the value on the left is a single item from the value on the right-hand-side, and yet the last value (the second `_` in your suggestion) is assigned all of the remaining items.  I was hoping for a way to retain that information for future readers/copiers in case they attempted to make adjustments that wouldn't work as they expect.

Reply:
Ah ok, you learn something new every day. ;) I might use an underscore just for the first element then. But as I said above, not a big deal.

Reply:
> and yet the last value (the second _ in your suggestion) is assigned all of the remaining items

Technically, no. It would be assigned the remaining items if you were using `...rest`. You can check:

```javascript
[_, two, _] = [1,2,3,4,5]
console.log(`${_}`) // 3

[_, two, ..._] = [1,2,3,4,5]
console.log(`${_}`) // 3,4,5
```

It's the same as in Python:

```python
(one, two, *threefourfive) = (1, 2, 3, 4, 5)
```

Reply:
Ok, yep - I misunderstood JavaScript's unpacking behaviour.  I'll adjust this to use meaningful-and-underscore-prefixed variable names, and with unpacking for the remainder items.